prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -2071,6 +2071,74 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                             }
                         }
 
+                        "core::slice::<impl [T]>::len" => {
+                            trace!("full_func_proc_name: {}", full_func_proc_name);
+                            trace!("self_ty: {:?}", self_ty);
+                            trace!("def_id: {:?}", def_id);
+                            trace!("substs: {:?}", substs);
+                            trace!("args: {:?}", args);
+                            trace!("destination: {:?}", destination);
+                            trace!("ty: {:?}", ty);
+
+                            assert!(args.len() == 1, "unexpected args to slice::len(): {:?}", args);
+                            let slice_operand = self.mir_encoder.encode_operand_expr(&args[0])
+                                .with_span(span)?;
```

Comment:
Add `.run_if_err(|| cleanup(&self))?` to call the cleanup closure before returning an error. (Here and for all the `?` below.)

Reply:
moved the whole block into a separate function, so this should only be needed once on the function call.